src/components/table/column-picker/column-picker.component.spec.ts

```diff
@@ -0,0 +1,216 @@
+        <ux-column-picker
+            [(selected)]="selected"
+            [(deselected)]="deselected"
+            [locked]="locked"
+            [groups]="groups">
+        </ux-column-picker>
+    `
+})
+export class ColumnPickerTestComponent {
+    selected: ReadonlyArray<string> = [
+        'Type',
+        'Date',
+        'Requested by',
+        'Status',
+        'Completion'
+    ];
+
+    locked: ReadonlyArray<string> = ['ID'];
+
+    deselected: Array<string | ColumnPickerGroupItem> = [
+        { group: 'Metadata', name: 'Department' },
+        { group: 'Metadata', name: 'Author' },
+        { group: 'Metadata', name: 'Category' },
+        { group: 'Metadata', name: 'Date Created' },
+        { group: 'Metadata', name: 'Date Modified' },
+        'Document ID',
+        'Flag',
+        'From',
+        'Icon',
+        'Importance',
+        'Location',
+        'Location ID',
+        'Message',
+        { group: 'Metadata', name: 'Organization' },
+        'Time',
+        'Time Created',
+        'Time Modified',
+        'Work Completed'
+    ];
+
+    groups: ColumnPickerGroup[] = [
+        { name: 'Metadata', expanded: true }
+    ];
+}
+
+describe('Column Picker Component', () => {
+    let component: ColumnPickerTestComponent;
+    let fixture: ComponentFixture<ColumnPickerTestComponent>;
+    let nativeElement: HTMLElement;
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            imports: [TableModule],
+            declarations: [ColumnPickerTestComponent]
+        }).compileComponents();
+    }));
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(ColumnPickerTestComponent);
+        component = fixture.componentInstance;
+        nativeElement = fixture.nativeElement;
+        fixture.detectChanges();
+    });
+
+    it('should display the correct number of deselected columns', async () => {
+        expect(component.deselected.length).toEqual(18);
+        let deselected = nativeElement.querySelector('.column-picker-list').querySelectorAll('.column-picker-list-item').length;
+        expect(deselected).toEqual(18);
+
+        component.deselected = ['Document ID', 'Flag', 'From', 'Icon'];
+
+        fixture.detectChanges();
+        await fixture.whenStable();
+
+        expect(component.deselected.length).toEqual(4);
+        deselected = nativeElement.querySelector('.column-picker-list').querySelectorAll('.column-picker-list-item').length;
+        expect(deselected).toEqual(4);
+    });
+
+    it('should not sort the deselected columns when the sort input is undefined', () => {
+        expect(component.deselected[0]).toEqual({ group: 'Metadata', name: 'Department' });
+    });
+
+    it('should update the number of deselected columns available to select', async () => {
+        let numberOfDeselectedColumns = nativeElement.querySelector('.column-picker-stats');
+        expect(numberOfDeselectedColumns.textContent.trim()).toEqual('0 of 18 selected');
```

Comment:
I think it would be reasonable to add this check into the first test since it's related.